### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by james-d-mitchell**
Some cleanups, there's one fixme, which is that in the current line

dev/releases/update_website.py:200:

there's no function named `extract_tarball` in any of these scripts, so this line would fail if executed. I haven't tested these changes at all.

### CODE REVIEW

File: dev/releases/generate_package_release_notes.py [hunk 1]
```diff
@@ -90,12 +103,15 @@ def main(old_gap_version, new_gap_version):
...
     if len(updated) > 0:
-        print(f"""
+        print(
+            f"""
 #### Updated packages redistributed with GAP
 
-The GAP {new_gap_version} distribution contains {len(new_json)} packages, of which {len(updated)} have been
-updated since GAP {old_gap_version}. The full list of updated packages is given below:
-""".lstrip())
+The GAP {new_gap_version} distribution contains {len(new_json)} packages, of
```
Comment (fingolfin) [inline dev/releases/generate_package_release_notes.py]: This shouldn't be rewrapped like that, as it changes the line breaks in the final file: in the generated file, `{len(new_json)}` is replaced by 3 digits, and `{len(updated)}` by 1 to 3, etc..

File: dev/releases/make_archives.py [hunk 2]
```diff
@@ -54,70 +68,92 @@
...
 
 # extract commit_date with format YYYY-MM-DD
 # TODO: is that really what we want? Or should it be the date this
 # script was run? Or for releases, perhaps use the TaggerDate of the
 # release tag (but then we need to find and process that tag)
-commit_date = subprocess.run(["git", "show", "-s", "--format=%as"],
-                             check=True, capture_output=True, text=True)
+commit_date = subprocess.run(
+    ["git", "show", "-s", "--format=%as"], check=True, capture_output=True, text=True
+)
```
Comment (fingolfin) [inline dev/releases/make_archives.py:88]: I don't mind much, but personally I find the new version harder to read than the old :-)

File: dev/releases/make_archives.py [hunk 3]
```diff
@@ -54,70 +68,92 @@
...
 notice("Processing exported content")
-manifest_list = [] # collect names of assets to be uploaded to GitHub release
+manifest_list = []  # collect names of assets to be uploaded to GitHub release
 
 # download package distribution
-notice("Downloading package distribution")   # ... outside of the directory we just created
-download_with_sha256(PKG_BOOTSTRAP_URL+"package-infos.json.gz", tmpdir+"/"+"package-infos.json.gz")
+notice(
+    "Downloading package distribution"
+)  # ... outside of the directory we just created
```
Comment (fingolfin) [inline dev/releases/make_archives.py]: Perhaps better:
```suggestion
# outside of the directory we just created
notice("Downloading package distribution")
```

File: dev/releases/make_archives.py [hunk 4]
```diff
@@ -226,29 +271,30 @@ def make_and_record_archive(name, compression, root_dir, base_dir):
     else:
         error(f"unknown compression type {compression} (not gztar or zip)")
 
-    filename = f"{name}{ext}"
-    notice(f"Creating {filename}")
+    fname = f"{name}{ext}"
```
Comment (fingolfin) [inline dev/releases/make_archives.py]: Maybe also just
```suggestion
    fname = name + ext
```

File: dev/releases/update_website.py [hunk 5]
```diff
@@ -176,16 +197,19 @@ def download_and_extract_json_gz_asset(asset_name, dest):
         tarball = f"gap-{version}-core.tar.gz"
         download_asset_by_name(tarball, tmpdir)
         with utils.working_directory(tmpdir):
+            # FIXME extract_tarball is not defined
```
Comment (fingolfin) [inline dev/releases/update_website.py:200]: Huh? It is defined in line 83 of this file?

Comment (fingolfin) [reply dev/releases/update_website.py:200]: ah you accidentally renamed it to `extract_tball` :-)

File: dev/releases/update_website.py [hunk 6]
```diff
@@ -65,40 +70,47 @@
...
 
     with utils.working_directory(writedir):
         notice(f"Downloading {url} to {writedir} . . .")
         utils.download_with_sha256(url, asset_name)
 
-def extract_tarball(tarball):
-    notice(f"Extracting {tarball} . . .")
-    with tarfile.open(tarball) as tar:
+
+def extract_tball(tball):
```
Comment (fingolfin) [inline dev/releases/update_website.py]:
```suggestion
def extract_tarball(tarball):
```
and then also change `tball` below...

File: dev/releases/utils.py [hunk 7]
```diff
@@ -15,42 +15,54 @@
...
+
 # check for uncommitted changes
 def is_git_clean():
-    res = subprocess.run(["git", "update-index", "--refresh"])
+    res = subprocess.run(["git", "update-index", "--refresh"], check=False)
     if res.returncode == 0:
-        res = subprocess.run(["git", "diff-index", "--quiet", "HEAD", "--"])
+        res = subprocess.run(
+            ["git", "diff-index", "--quiet", "HEAD", "--"], check=False
+        )
```
Comment (fingolfin) [inline dev/releases/utils.py:57]: What are you using to format the code? I just tried `black` and it doesn't touch `is_git_clean` for me

Comment (james-d-mitchell) [reply dev/releases/utils.py:57]: I used black, sorry, I didn't really mean to format everything again. Note that here I also added `check=False`, which might make the difference.

File: dev/releases/update_website.py (hunk 5 above)
Comment (james-d-mitchell) [reply dev/releases/update_website.py:200]: Ooops, my bad

File: dev/releases/generate_package_release_notes.py (hunk 1 above)
Comment (james-d-mitchell) [reply dev/releases/generate_package_release_notes.py]: I've undone that too

**Pull request closed without merging**